scipy/special/specfun.c

```diff
@@ -0,0 +1,8417 @@
+        }
+        if (id > -1) {
+            specfun_rmn2so(m, n, c, x, cv, kd, df, r2f, r2d);
+        }
+    }
+    free(df);
+    return;
+}
+
+
+void specfun_sckb(int m, int n, double c, double *df, double *ck) {
+
+    // ======================================================
+    // Purpose: Compute the expansion coefficients of the
+    //          prolate and oblate spheroidal functions
+    // Input :  m  --- Mode parameter
+    //          n  --- Mode parameter
+    //          c  --- Spheroidal parameter
+    //          DF(k) --- Expansion coefficients dk
+    // Output:  CK(k) --- Expansion coefficients ck;
+    //                    CK(1), CK(2), ... correspond to
+    //                    c0, c2, ...
+    // ======================================================
+
+    int i, ip, i1, i2, k, nm;
+    double reg, fac, sw, r, d1, d2, d3, sum, r1;
+
+    if (c <= 1.0e-10) {
+        c = 1.0e-10;
+    }
+    nm = 25 + (int)(0.5 * (n - m) + c);
+    ip = (n - m) % 2;
+    reg = ((m + nm) > 80 ? 1.0e-200 : 1.0);
+    fac = -pow(0.5, m);
+    sw = 0.0;
+
+    for (k = 0; k < nm; k++) {
+        fac = -fac;
+        i1 = 2 * k + ip + 1;
+        r = reg;
+
+        for (i = i1; i <= i1 + 2 * m - 1; i++) {
+            r *= i;
+        }
+        i2 = k + m + ip;
+        for (i = i2; i <= i2 + k - 1; i++) {
+            r *= (i + 0.5);
+        }
+        sum = r * df[k];
+        for (i = k + 1; i <= nm; i++) {
+            d1 = 2.0 * i + ip;
+            d2 = 2.0 * m + d1;
+            d3 = i + m + ip - 0.5;
+            r *= d2 * (d2 - 1.0) * i * (d3 + k) / (d1 * (d1 - 1.0) * (i - k) * d3);
+            sum += r * df[i];
+            if (fabs(sw - sum) < fabs(sum) * 1.0e-14) { break; }
+            sw = sum;
+        }
+        r1 = reg;
+        for (i = 2; i <= m + k; i++) { r1 *= i; }
+        ck[k] = fac * sum / r1;
+    }
+}
+
+
+void specfun_sdmn(int m, int n, double c, double cv, int kd, double *df) {
+
+    // =====================================================
+    // Purpose: Compute the expansion coefficients of the
+    //          prolate and oblate spheroidal functions, dk
+    // Input :  m  --- Mode parameter
+    //          n  --- Mode parameter
+    //          c  --- Spheroidal parameter
+    //          cv --- Characteristic value
+    //          KD --- Function code
+    //                 KD=1 for prolate; KD=-1 for oblate
+    // Output:  DF(k) --- Expansion coefficients dk;
+    //                    DF(1), DF(2), ... correspond to
+    //                    d0, d2, ... for even n-m and d1,
+    //                    d3, ... for odd n-m
+    // =====================================================
+
+    int nm, ip, k, kb;
+    double cs, dk0, dk1, dk2, d2k, f, fs, f1, f0, fl, f2, su1,\
+           su2, sw, r1, r3, r4, s0;
+
+    nm = 25 + (int)(0.5 * (n - m) + c);
+
+    if (c < 1e-10) {
+        for (int i = 1; i <= nm; ++i) {
+            df[i-1] = 0.0;
+        }
+        df[(n - m) / 2 + 1] = 1.0;
```

Comment:
```suggestion
        df[(n - m) / 2] = 1.0;
```

https://github.com/scipy/scipy/blob/829d0dd950d63caece04526b3a617bbe2bc24023/scipy/special/specfun/specfun.f#L10302-L10304

off by one due to 1 based 0 based confusion